### DISCUSSION TITLE
Genetic algorithm in netket

### DISCUSSION THREAD

**Discussion by Ashu-20** [netket/netket#1450] [Q&A] [answered]
Can I use genetic algorithm for optimization in netket? If yes how?
For reference I am trying to optimize the parameter of feed forward neural network used to calculate the ground state of Heisenberg model using genetic algorithm.

**Comment by gcarleo** [comment, marked as answer]
No, not implemented. I advice against it

**Reply by Ashu-20** [reply]
Can you please explain why you are against it?

**Reply by gcarleo** [reply]
state of the art deep learning models are trained with gradients, these are consistently better than genetic algorithms. I am not aware of a single instance where genetic algorithms are a superior alternative to gradients, especially in a context such as deep learning where you have gradients (more or less) for free!

**Reply by PhilipVinc** [reply]
(And genetic algorithms are in general used in contexts where you either have few parameters or can't compute the gradients. As Giuseppe says, if you have them for almost free, not using them is not exactly a 'computationally efficient' choice.)
